Tests/TuistGeneratorTests/Generator/ConfigGeneratorTests.swift

```diff
@@ -159,7 +159,7 @@ final class ConfigGeneratorTests: TuistUnitTestCase {
 
     func test_generateTestTargetConfiguration_macOS() throws {
```

Comment:
There are some changes in `ConfigGeneratorTests` that would benefit from some unit tests here to prevent regressions:
- The right `TARGETED_DEVICE_FAMILY` build setting is set based on the target destinations.
- The build settings `SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD`, `SUPPORTS_XR_DESIGNED_FOR_IPHONE_IPAD`, `SUPPORTS_MACCATALYST`, and `DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER` are set accordingly.
- The right `_DEPLOYMENT_TARGET` environment variables are set.